File: notes-sharing-app-be/internal/models/note.go
package models

import "time"

type Note struct {
	ID        string    `json:"id"`
	Title     string    `json:"title" validate:"required,min=5"`
	Content   string    `json:"content" validate:"required,min=5"`
	UserID	  string	`json:"user_id" validate:"required"`
	Username  string    `json:"username"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: notes-sharing-app-fe/src/types/note.ts
export interface Note {
  id: string;
  title: string;
  content: string;
  user_id?: number;
  username?: string;
  created_at: string;
  updated_at: string;
}

export type NotesResponse = {
  status: number;
  message: string;
  data?: {
    limit: number;
    notes: Note[];
    page: number;
    total: number;
  };
};

export type NoteResponse = {
  status: number;
  message: string;
  data?: Note;
};
